lib/js-yaml/dumper.js

```diff
@@ -291,12 +284,14 @@ function chooseScalarStyle(string, singleLineOnly, indentPerLevel, lineWidth, te
       } else if (!isPrintable(char)) {
         return STYLE_DOUBLE;
       }
-      plain = plain && isPlainSafe(char);
+      if (!isPlainSafe(char)) {
+        plain = plain && !isWhitespace(string.charCodeAt(i + 1));
+      }
```

Comment:
This is a very cryptic way to express "if plainsafe and next is not whitespace". Very deficult to understand real intent. Please, reorganize somehow.